app/models/project_step.rb

```diff
@@ -80,4 +78,49 @@ def status
   def display_date
     I18n.l (self.completed_date || self.scheduled_date), format: :long
   end
+
```

Comment:
Lines 82-125 are stubbed. Some of these methods need to be revised with the actual logic that will be used. Some need to be removed altogether.